dotCMS/src/main/java/com/dotcms/cms/login/LoginServiceAPIImpl.java

```diff
@@ -0,0 +1,537 @@
+        _doLogoutInternal(req, res);
+        
+        
+        event = new LoginEvent(this, LoginEvents.POST_LOGOUT, null, null, req, res);
+        for (LoginEventListener listener : _listeners) {
+            listener.loginEventReceived(event);
+        }
+    }
+
+
+
+    private void _doLogoutInternal(final HttpServletRequest req, final HttpServletResponse res) {
+
+        final HttpSession session = req.getSession(false);
+
+        log.debug("Logout - Set expire cookies");
+        com.dotmarketing.util.CookieUtil.setExpireCookies(req, res);
+
+        if (null != session) {
+
+            log.debug("Logout - Events Processor Pre Logout events.");
+
+
+            log.debug("Logout - Invalidating http session...");
+
+            session.invalidate();
+
+            log.debug("Logout - Events Processor Post Logout events.");
+
+        }
+    }
+
+
+
+    /**
+     * Generates the JWT and its respective cookie based on the following criteria:
+     * <ul>
+     * <li>Information from the user: ID, associated company.</li>
+     * <li>The "Remember Me" option being checked or not.</li>
+     * </ul>
+     * The JWT will allow the user to access the dotCMS back-end even though their session has
+     * expired. This way, they won't need to re-authenticate.
+     *
+     * @param req - The {@link HttpServletRequest} object.
+     * @param res - The {@link HttpServletResponse} object.
+     * @param user - The {@link User} trying to log into the system.
+     * @param maxAge - The maximum days (in milliseconds) that the JWT will live. If the "Remember
+     *        Me" option is checked, the token will live for many days (check its default value).
+     *        Otherwise, it will only live during the current session.
+     * @throws PortalException The specified user could not be found.
+     * @throws SystemException An error occurred during the user ID encryption.
+     */
+    private void processJsonWebToken(final HttpServletRequest req, final HttpServletResponse res,
+                    final User user, final int maxAge) throws PortalException, SystemException {
+
+        final String jwtAccessToken = this.jsonWebTokenUtils.createToken(user, maxAge);
+        createJsonWebTokenCookie(req, res, jwtAccessToken, Optional.of(maxAge));
+    }
+
+
+
+    @Override
+    public void doRememberMe(final HttpServletRequest req, final HttpServletResponse res,
+                    final User user, final boolean rememberMe) {
+
+        int jwtMaxAge = rememberMe ? Config.getIntProperty(JSON_WEB_TOKEN_DAYS_MAX_AGE,
+                        JSON_WEB_TOKEN_DAYS_MAX_AGE_DEFAULT) : -1;
+
+        this.doRememberMe(req, res, user, jwtMaxAge);
+    } // doRememberMe.
+
+    @Override
+    public void doRememberMe(final HttpServletRequest req, final HttpServletResponse res,
+                    final User user, final int maxAge) {
+
+        try {
+
+            this.processJsonWebToken(req, res, user, maxAge);
+        } catch (Exception e) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)